toonz/sources/toonz/cellselection.cpp

```diff
@@ -453,6 +454,19 @@ class RenumberUndo final : public TUndo {
         std::vector<TXshCell> cells(n);
         xsh->getCells(r0, c, n, &cells[0]);
         for (int i = 0; i < n; i++) {
+          // check the sub xsheets too
+          if (!cells[i].isEmpty() &&
+              cells[i].m_level->getType() == CHILD_XSHLEVEL) {
+            TXshChildLevel *level = cells[i].m_level->getChildLevel();
+            // make sure we haven't already checked the level
+            if (level &&
+                std::find(childLevels.begin(), childLevels.end(), level) ==
+                    childLevels.end()) {
```

Comment:
@turtletooth Sorry for the late notice, but I found a ( minor ) bug here.
In this line you are checking whether the subxsheet level is already renumbered by using the list `childLevels` .
However, there may be a case that the same subxsheet level is used in different hierarchies of subxsheets.
For example like this:
![image](https://user-images.githubusercontent.com/17974955/32939404-aeb60e92-cbc2-11e7-9b7c-166d7494429c.png)

In the above example, the subxsheet `sub_B` is used both in the parent xhseet and in the subxsheet `sub_A` .
When renumbering the level `A` , this condition cannot catch such usage and `sub_B` will be renumbered twice.

I think for the renumbering of subxsheet it would better to share the list among the hierarchy.